### PR TITLE
feature: Add new encoding runlength v3 using Longs instead of Integers

### PR DESCRIPTION
**Pull request by astubbs**
From the javadoc:

> Uses {@link Long} for it's underlying structure (in order to support output encoding option using Longs), in order to  support the worst case scenarios without running out of run length, as {@link Integer#MAX_VALUE} of 2,147,483,647 (2 billion records) is not completely inconceivable, whereas {@link Long#MAX_VALUE} of 9,223,372,036,854,775,807 (~9 quintillion / 9 x 10^18) is.

### Checklist

- [ ] Documentation (if applicable)
- [ ] Changelog

Blocked by:
- #439

### CODE REVIEW

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/offsets/RunLengthEncoder.java [hunk 1]
```diff
@@ -75,19 +80,24 @@ public byte[] serialise() throws EncodingNotSupportedException {
...
         };
-        ByteBuffer runLengthEncodedByteBuffer = ByteBuffer.allocate(runLengthEncodingIntegers.size() * entryWidth);
+        ByteBuffer runLengthEncodedByteBuffer = ByteBuffer.allocate(runLengthEncodingLongs.size() * entryWidth);
 
-        for (final Integer runLength : runLengthEncodingIntegers) {
+        for (final Long runLength : runLengthEncodingLongs) {
             switch (version) {
                 case v1 -> {
                     final short shortCastRunlength = runLength.shortValue();
+//                    final short shortCastRunlength = MathUtils.toShortExact(runLength);
```
Comment (astubbs) [inline parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/offsets/RunLengthEncoder.java:91]: use

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/offsets/OffsetRunLength.java [hunk 2]
```diff
@@ -3,21 +3,23 @@
...
 import java.util.List;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicInteger;
-import java.util.function.Supplier;
+import java.util.function.BooleanSupplier;
 
+/**
+ * @author Antony Stubbs
+ */
+// todo rename - resolve name against RunLengthEncoder
```
Comment (astubbs) [inline parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/offsets/OffsetRunLength.java:22]: rename?

File: parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/offsets/RunLengthEncoder.java [hunk 3]
```diff
@@ -47,6 +50,7 @@ protected OffsetEncoding getEncodingType() {
         return switch (version) {
             case v1 -> RunLength;
             case v2 -> RunLengthV2;
+            case v3 -> RunLengthV3;
```
Comment (astubbs) [inline parallel-consumer-core/src/main/java/io/confluent/parallelconsumer/offsets/RunLengthEncoder.java:53]: write unit test to exercise the expanded range?

**Review by astubbs — COMMENTED** [review]
just needs a test

**Review by astubbs — COMMENTED** [review]
....

**Comment by eddyv** [comment]
Closing - Stale.

**Pull request closed without merging**